lib/pages/register-page.dart

```diff
@@ -0,0 +1,220 @@
+                              borderSide: BorderSide(color: Colors.transparent),
+                            ),
+                            filled: true,
+                            fillColor: Colors.grey[300],
+                            labelText: getString(context, 'confirm_password_label'),
+                            labelStyle: TextStyle(
+                              fontWeight: FontWeight.bold,
+                            )),
+                        controller: _confirmPasswordController,
+                        obscureText: true,
+                        onChanged: (text) => handleSubmittedConfirmPassword(text),
+                      ),
+                    ),
+                    Positioned(
+                      top: (MediaQuery.of(context).size.height / 2) + 90,
+                      left: (MediaQuery.of(context).size.width / 2) - 65,
+                      child: RaisedButton(
+                        key: Key("create_account_button"),
+                        onPressed: () {
+                          if (_formKey.currentState.validate()) {
+                            createAccount();
+                          }
+                        },
+                        child: Text(getString(context, 'create_account')),
+                      ),
+                    ),
+                  ],
+                ),
+              ),
+              Positioned(
+                top: 70,
+                left: (MediaQuery.of(context).size.width / 2) - 120,
+                child: 
+                  Text(getString(context, 'register_page_text'), 
+                    style: TextStyle(
+                      fontSize: 25,
+                      fontWeight: FontWeight.bold,
+                    ),
+                  ),
+              ),
+              Positioned(
+                right: 10,
+                bottom: 10,
+                child: FlatButton(
+                  color: Colors.transparent,
+                  onPressed: () => openAuthenticationPage(context),
+                  child: Text(getString(context, 'authentication_page_button'), 
+                    style: TextStyle(
+                      color: Colors.blue,
+                    ),
+                  ),
+                ),
+              ),
+            ],
+          ),
+        ), 
+      ),
+    );
+  }
+
+  Future<void> createAccount() async {
+    String result = await AuthenticationService().register(_email, _password);
+    switch (result) {
+      case "success":
+        Fluttertoast.showToast(msg: getString(context, 'successful_registration'));
+        openAuthenticationPage(context);
+        break;
+      case "format_error":
+        Fluttertoast.showToast(msg: getString(context, 'format_error'));
+        break;
+      case "email_already_exist":
+        Fluttertoast.showToast(msg: getString(context, 'email_already_exist'));
+        break;
+    }
+  }
+
+  void openAuthenticationPage(context) {
+    Navigator.of(context).pushNamed('/');
```

Comment:
you should cleanup the navigation stack when returnin to the auth page

try Navigator.popUntil(context, ModalRoute.withName('/'));